SemanticMediaWiki.php

```diff
@@ -24,129 +20,64 @@
-	SemanticMediaWiki::onExtensionFunction();
-};
-
 /**
- * @codeCoverageIgnore
+ * THIS IS A TEMPORARY HACK to get around the #1699 issue in connection with the
+ * tarball release that conflicts with the Composer autoloading when invoked
+ * via the LocalSettings.
+ *
+ * By the time `extension.json` is used, the content from load.php is to be moved
+ * into this file.
  */
-class SemanticMediaWiki {
-
-	/**
-	 * As soon as Composer is autoloading this file, start the init process for some
-	 * components.
-	 *
-	 * @since 2.4
-	 */
-	public static function initExtension() {
-
-		if ( is_readable( __DIR__ . '/vendor/autoload.php' ) ) {
-			include_once __DIR__ . '/vendor/autoload.php';
-		}
-
-		// Registration of the extension credits, see Special:Version.
-		$GLOBALS['wgExtensionCredits']['semantic'][] = array(
-			'path' => __FILE__,
-			'name' => 'Semantic MediaWiki',
-			'version' => SMW_VERSION,
-			'author' => array(
-				'[http://korrekt.org Markus Krötzsch]',
-				'[https://www.mediawiki.org/wiki/User:Jeroen_De_Dauw Jeroen De Dauw]',
-				'James Hong Kong',
-				'[https://www.semantic-mediawiki.org/wiki/Contributors ...]'
-				),
-			'url' => 'https://www.semantic-mediawiki.org',
-			'descriptionmsg' => 'smw-desc',
-			'license-name'   => 'GPL-2.0+'
-		);
-
-		// A flag used to indicate SMW defines a semantic extension type for extension credits.
-		// @deprecated, removal in SMW 3.0
-		define( 'SEMANTIC_EXTENSION_TYPE', true );
-
-		// Load class_alias
-		require_once __DIR__ . '/includes/Aliases.php';
-
-		// Load global constants
-		require_once __DIR__ . '/includes/Defines.php';
-
-		// Temporary measure to ease Composer/MW 1.22 migration
-		require_once __DIR__ . '/includes/NamespaceManager.php';
-
-		// Load global functions
-		require_once __DIR__ . '/includes/GlobalFunctions.php';
-
-		// Load default settings
-		require_once __DIR__ . '/SemanticMediaWiki.settings.php';
-
-		// Because of MW 1.19 we need to register message files here
-		$GLOBALS['wgMessagesDirs']['SemanticMediaWiki'] = $GLOBALS['smwgIP'] . 'i18n';
-		$GLOBALS['wgExtensionMessagesFiles']['SemanticMediaWiki'] = $GLOBALS['smwgIP'] . 'languages/SMW_Messages.php';
-		$GLOBALS['wgExtensionMessagesFiles']['SemanticMediaWikiAlias'] = $GLOBALS['smwgIP'] . 'languages/SMW_Aliases.php';
-		$GLOBALS['wgExtensionMessagesFiles']['SemanticMediaWikiMagic'] = $GLOBALS['smwgIP'] . 'languages/SMW_Magic.php';
-	}
-
-	/**
-	 * Setup and initialization
-	 *
-	 * @note $wgExtensionFunctions variable is an array that stores
-	 * functions to be called after most of MediaWiki initialization
-	 * has finalized
-	 *
-	 * @see https://www.mediawiki.org/wiki/Manual:$wgExtensionFunctions
-	 *
-	 * @since  1.9
-	 */
-	public static function onExtensionFunction() {
-
-		// 3.x reverse the order to ensure that smwgMainCacheType is used
-		// as main and smwgCacheType being deprecated with 3.x
-		$GLOBALS['smwgMainCacheType'] = $GLOBALS['smwgCacheType'];
+require_once __DIR__ . "/load.php";
```

Comment:
You sure the issue is fixed like this? That file still defines symbols and has side effects